[Actions] Simplify emitter schema model

snoe opened the pull request:
Addresses #24529

dashboard_emitter.dashboard_id and emitter_action.action_id columns
moved onto emitter table. Removed code that had to handle the table
inheritence of DashboardEmitter/CardEmitter, allowed tests to use
with-temp directly rather than with-emitter.

card_emitter was dropped since we don't have a solid use case for it
yet. Actions may be attached to models in the future but it's unclear
if a card_emitter is the right tool for the job at that point.

emitter_action was dropped since emitters just have a singular action
and the join table was unecessary.

Creates new migration steps since pod-writeback was deployed to a demo
site and we want to keep that working.

dpsutton committed:
Handle flakiness with geojson java.net.UnknownHostException errors (#24523)

* Handle flakiness with geojson java.net.UnknownHostException errors

In CI seems like we are getting errant errors:

```clojure
geojson.clj:62
It validates URLs and files appropriately
http://0xc0000200
expected: (valid? geojson)
  actual: #error {
 :cause "Invalid IP address literal: 0xc0000200"
 :via
 [{:type clojure.lang.ExceptionInfo
   :message "Invalid GeoJSON file location: must either start with http:// or https:// or be a relative path to a file on the classpath. URLs referring to hosts that supply internal hosting metadata are prohibited."
   :data {:status-code 400, :url "http://0xc0000200"}
   :at [metabase.api.geojson$valid_url_QMARK_ invokeStatic "geojson.clj" 62]}
  {:type java.net.UnknownHostException
   :message "0xc0000200"
   :at [java.net.InetAddress getAllByName "InetAddress.java" 1340]}
  {:type java.lang.IllegalArgumentException
   :message "Invalid IP address literal: 0xc0000200"
   :at [sun.net.util.IPAddressUtil validateNumericFormatV4 "IPAddressUtil.java" 150]}]
```

Not clear if this change has a hope of fixing it: if it doesn't resolve
once its possible it is cached somewhere in the network stack, or it
won't resolve if you ask again.

But gonna give it a shot.

Set the property `"networkaddress.cache.negative.ttl"` to `"0"`

> networkaddress.cache.negative.ttl (default: 10)
>    Indicates the caching policy for un-successful name lookups from the name service. The value is specified as an integer to indicate the number of seconds to cache the failure for un-successful lookups.

>    A value of 0 indicates "never cache". A value of -1 indicates "cache forever".

From
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/net/InetAddress.html
in the hopes that we can try multiple times. Restores the original value
after the test completes so we don't inadvertently change behavior
elsewhere.

If we get an error of java.net.UnknownHostException we try again if we
have attempts remaining. If we get a boolean it means the ip resolution
worked so we can rely on the response (checking if it resolves locally
or not)

* add a delay

* comment out test

snoe commented on resources/migrations/000_migrations.yaml:
```diff
@@ -12286,6 +12286,113 @@ databaseChangeLog:
+      comment: Added 0.45.0 - writeback drop emitter_action
+      changes:
+        - dropTable:
+            tableName: emitter_action
```
Question: These drop tables can't rollback manually, is that a concern?

snoe replied:
Question: should the v44 writeback migrations move to v45? They were not released as part of 44 but the writeback demo site did run them.

camsaul replied:
Yeah, we should move them all to 45. You can add conditions to skip them if their old numbers have already been ran. That can be done as a separate PR tho. It might actually make sense to tackle that first and then rebase this PR off of that one

camsaul commented:
I think it might sense to leave `card_emitter` in for right now since it seems like we really will probably want to be using Models as emitters in the future. If we didn't have it yet I wouldn't say we need to add it now, but seeing how we already have it and have some tests around it we may as well leave it in. If anything it will keep us from writing too much code that won't work with different types of Emitters as we move forward

camsaul committed:
Move writeback migrations to 45

camsaul committed:
Empty commit to trigger GitHub Actions

snoe committed:
[Actions] Simplify emitter schema model

emitter_action was dropped since emitters just have a singular action
and the join table was unecessary.

emitter_action.action_id columns moved onto emitter table.

Dropped CardEmitter and DashboardEmitter pre-insert, pre-update,
pre-delete since they were used in tests only and normal operation
would see the emitter inserted first.

Since previous code may have 'orphaned' emitters without an action, we
delete emitters without action to be safe.

snoe committed:
Merge remote-tracking branch 'origin/pod-writeback' into move-writeback-migrations-to-45

snoe committed:
Merge branch 'move-writeback-migrations-to-45' into writeback-schema-refactor-emitter

snoe committed:
Merge remote-tracking branch 'origin/pod-writeback' into writeback-schema-refactor-emitter
